From 88c9a62b6d3e4d059bde39d31cf93abfe7133863 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Wed, 3 Apr 2024 17:04:13 +0200
Subject: [PATCH 1084/1141] arm64: dts: st: Add lpuart1 in the stm32mp25x soc

Add lpuart1 node in stm32mp251.dtsi. Use the same compatible than
classical USART "st,stm32h7-uart". And same properties interrupt, clocks
dma and feature-domains.

Change-Id: Id8daaa75d69e82e8d6c9c438e77dd3bfcc8c903e
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/363099
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -1940,6 +1940,18 @@
 				status = "disabled";
 			};
 
+			lpuart1: serial@46030000 {
+				compatible = "st,stm32h7-uart";
+				reg = <0x46030000 0x400>;
+				interrupts-extended = <&exti2 26 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_LPUART1>;
+				dmas = <&hpdma 166 0x20 0x10012 0x0>,
+				       <&hpdma 167 0x20 0x3021 0x0>;
+				dma-names = "rx", "tx";
+				feature-domains = <&rifsc STM32MP25_RIFSC_LPUART1_ID>;
+				status = "disabled";
+			};
+
 			i2c8: i2c@46040000 {
 				compatible = "st,stm32mp25-i2c";
 				reg = <0x46040000 0x400>;
